packages/blocks/src/api/registration.js

```diff
@@ -608,6 +608,18 @@ export function isReusableBlock( blockOrType ) {
 	return blockOrType?.name === 'core/block';
 }
 
+/**
+ * Determines whether or not the given block is a navigation block. This is a
+ * special block type that can used to point to a wp_navigation post type.
+ *
+ * @param {Object} blockOrType Block or Block Type to test.
+ *
+ * @return {boolean} Whether the given block is a navigation block.
+ */
+export function isNavigationBlock( blockOrType ) {
+	return blockOrType?.name === 'core/navigation';
+}
```

Comment:
I know there's already `isTemplatePart` and `isReusableBlock`, but IMO those are a mistake. The blocks package should be agnostic to the types in the block library. I don't think more of these functions should be added.

Reply:
I agree. Probably not worth introducing a new API method just for this check.

@draganescu, what do you think if we update edit locking checks to something like this?

```js
// A module-level const
const ALLOWS_EDIT_LOCKING = [ 'core/block', 'core/navigation' ];


// Then in `mapSelect` function
return {
  ...
  allowsEditLocking: ALLOWS_EDIT_LOCKING.includes( blockName ),
};
```

Reply:
Sure thing!

Reply:
Solved in [30d5d7e](https://github.com/WordPress/gutenberg/pull/44739/commits/30d5d7ed15a9a597dd2e6041cf89cf71c6cb9971)